build-tools/automation/build.groovy

```diff
@@ -10,12 +10,24 @@ def isPr = false                // Default to CI
 
 def buildTarget = 'jenkins'
 
-def stageWithTimeout(stageName, timeoutValue, timeoutUnit, directory, fatal, Closure body) {
+def stageWithTimeout(stageName, timeoutValue, timeoutUnit, directory, fatal, ctAttempts, Closure body) {
     try {
         stage(stageName) {
-            timeout(time: timeoutValue, unit: timeoutUnit) {
-                dir(directory) {
-                    body()
+            def retryAttempt = 0
+            def waitSecondsBeforeRetry = 15
+            retry(ctAttempts) {
```

Comment:
Trying to follow how this works, does passing a `0` here mean it still will run once?

Reply:
Short answer: Passing 0 or 1 results in the body being executed one time. Perhaps to avoid confusion (read on) we could use 1 as the default? (I've kept the default as 0 for now)
The semantics of the retry statement isn't what I'd expect, but I may be splitting hairs here. I would expect the retry count passed to the statement to be the additional retries that are attempted after the initial attempt. So a value of 2 would execute the body once and then up to 2 additional times if it fails. However, this isn't how it works. Instead the number of retries passed to the retry statement actually means the number of attempts it is going to make. So a value of 2 means that there is an initial attempt followed by 1 retry if that attempt fails.
Therefore I was intentional in naming this `ctAttempts` and not `ctRetries`